receiver/mongodbreceiver/client.go

```diff
@@ -109,6 +92,12 @@ func (c *mongodbClient) GetVersion(ctx context.Context) (*string, error) {
 		return nil, errors.New("unable to parse mongo version from server")
 	}
 
+	// TODO: get into version struct
+	version, err := version.NewVersion(v)
```

Comment:
Probably can just return here
```suggestion
	return version.NewVersion(v)
```

The debug log seems useful, but is actually probably more appropriate at the `start` level anyways.